frame/bridge/ethereum-bsc/src/lib.rs

```diff
@@ -354,6 +431,36 @@ pub mod pallet {
 			Ok(signers)
 		}
 
+		/// Verify single header
+		/// The header number should in the range [genesis_header.number, finalized_checkpoint.number + N]
+		/// Before the first call of verify_and_update_authority_set extrinsic, genesis_header == finalized_checkpoint
+		pub fn verify_header(header: &BSCHeader) -> DispatchResult {
+			let cfg = T::BSCConfiguration::get();
+			// ensure the number is in the range
+			let round_key = header.number / cfg.epoch_length;
+			ensure!(
+				<AuthoritiesOfRound<T>>::contains_key(round_key),
+				// it could be the signer which signed your header has not been finialized yet
+				// or your header.number is less than the genesis header number
+				<Error::<T>>::RidiculousNumber
+			);
+			Self::contextless_checks(&cfg, header)?;
+
+			// get index vec
+			let authorities_of_round = <AuthoritiesOfRound<T>>::get(round_key);
+			// get all authorities
+			let authorities = <Authorities<T>>::get();
+			// filter authorities of this round out
+			let signers = authorities_of_round
+				.iter()
+				.map(|i| authorities[*i as usize])
+				.collect::<Vec<Address>>();
```

Comment:
for each round, do we have the same authorities set? If not, we can't save the index instead of authorities.

Reply:
No, the authorities' slice may be(or not) growing in each round. So the authorities are not always the same.
But all qualified authorities will be appended to the set, so it's ok to have an index, not the address itself.
The only thing we should care about is to make sure the order is stable.

Reply:
ok I understand, authorities set is an array that saves all the authorities currently active or appeared in history. and the index to the authority is stable so we can find the authority by its index. Then for each round, we only need to record indexes of the authorities.